tests/test_xdp_container_restart.sh

```diff
@@ -0,0 +1,240 @@
+            entrypoint=""
+        fi
+        args+=("--entrypoint" "${entrypoint}")
+    fi
+
+    # Add tmpfs mount
+    args+=("--mount" "type=tmpfs,dst=/run")
+
+    # Add detached mode and image
+    args+=("-d" "${image_id}")
+
+    # Add command if provided
+    if [[ ${#command[@]} -gt 0 ]]; then
+        args+=("${command[@]}")
+    fi
+
+    local container_id
+    container_id=$("${OCI_BIN}" "${args[@]}")
+
+    if [[ -z "${container_id}" ]]; then
+        echo "Failed to run container ${container_name} with ${OCI_BIN}, got no container id" >&2
+        exit 1
+    fi
+
+    echo "${container_id}"
+}
+
+# Create the commit name from a provided name
+get_commit_name() {
+    local name="$1"
+    echo "${name}_commit"
+}
+
+# Function to execute command in container
+exec_in_container() {
+    local container_id="$1"
+    local cmd="$2"
+
+    local output
+    if ! output=$("${OCI_BIN}" exec "${container_id}" bash -c "${cmd}" 2>&1); then
+        echo "Command failed: ${output}" >&2
+        exit 1
+    fi
+
+    echo "${output}"
+}
+
+# Function to commit container to image
+commit_container() {
+    local container_id="$1"
+    local new_image_name="$2"
+
+    if ! "${OCI_BIN}" commit "${container_id}" "${new_image_name}"; then
+        echo "Failed to commit container" >&2
+        exit 1
+    fi
+}
+
+start_docker() {
+    echo "Starting docker service..."
+    if ! systemctl start docker; then
+        echo "Failed to start docker service" >&2
+        exit 1
+    fi
+
+    timeout="${DOCKER_TIMEOUT:-30}"
+    while [ "$timeout" -gt 0 ]; do
+        echo "Waiting for docker daemon to be ready... ($timeout)"
+        if docker info >/dev/null 2>&1; then
+            echo "Docker daemon is ready"
+            return
+        fi
+        sleep 1
+        ((timeout--))
+    done
+
+    echo "Docker daemon failed to become ready within ${DOCKER_TIMEOUT:-30} seconds" >&2
+    exit 1
+}
+
+# Main test function
+main() {
+    echo "Starting XDP container restart test..."
+
+    # Start OCI service (docker/podman)
+    if [[ "${OCI_BIN}" == "docker" ]]; then
+        if ! systemctl is-active --quiet docker; then
+            start_docker
+        fi
+    fi
+
+    # Set trap for cleanup
+    trap cleanup EXIT
+
+    # Clean up at start, just in case
+    cleanup
+
+    # Build container image
+    local current_image_id
+    current_image_id=$(build_container_image "${IMAGE_NAME}" "${CONTAINERFILE}")
+    current_container_name="${CONTAINER_NAME}"
+
+    # Test iterations: with and without commit
+    local iterations=(true false)
+    for should_commit in "${iterations[@]}"; do
+        echo "=== Iteration with should_commit=${should_commit} ==="
+
+        # Run new container
+        echo "Starting container ${current_container_name} with image ${current_image_id}"
+        local current_container_id
+        current_container_id=$(run_new_container "${current_container_name}" "${current_image_id}" "" "true" " " "sleep" "infinity")
```

Comment:
Why use "" (i.e., empty)? See my other comment in `run_new_container`.

Reply:
first one was for the port mapping (again, came from the rust -> shell conversion). I removed it. second one, see above